src/Diagnostics/average.jl

```diff
@@ -81,5 +84,6 @@ end
 
 function (avg::Average)(model)
     run_diagnostic(model, avg)
-    return avg.return_type(avg.result)
+    N, H = size(model.grid), halo_size(model.grid)
+    return avg.return_type(avg.result)[(avg.with_halos ? Colon() : d in avg.dims ? Colon() : (1+H[d]:N[d]+H[d]) for d in 1:3)...]
```

Comment:
This is probably overkill and could be much more readable if converted to an `if` statement with good comments...